app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Category;
+use App\Components\Recusive;
+
+class CategoryController extends Controller
+{
+    private $category;
+
+    public function __construct(Category $category) 
+    {
+        $this->category = $category;
+    } 
+
+    public function index()
+    {
+        $categories = Category::latest()->paginate(config('const.five'));
+
+        return view('admin.category.index', compact('categories'));
+    }
+
+    public function create()
+    {
+        $categoryOption = $this->getCategory($parentId = '');
+
+        return view('admin.category.add', compact('categoryOption'));
+    }
+
+    public function getCategory($parentId) 
+    {
+        $data = $this->category->all();
+        $recusive = new Recusive($data);
+        $categoryOption = $recusive->categoryRecusive($parentId);
+
+        return $categoryOption;
+    }
+
+    public function store(Request $request)
+    {
+        Category::create($request->all());
+
+        return redirect()->route('categories.index');
+    }
+
+    public function show($id)
+    {
+        //
+    }
+
+    public function edit($id)
+    {
+        $category = Category::findOrFail($id);
+        $categoryOption = $this->getCategory($category->parent_id);
+
+        return view('admin.category.edit', compact('category', 'categoryOption'));
+    }
+
+    public function update(Request $request, $id)
+    {
+        $category = Category::findOrFail($id);
```

Comment:
không tìm thấy id thì thế nào?

Reply:
nếu ko tìm thấy id thì nó sẽ trả về lỗi Not found ạ